### PR TITLE
disttask: add proto pkg for framework and example

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/41990

Problem Summary:

### What is changed and how it works?
- add proto pkg for disttask framework
- add example proto pkg
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/example/proto.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	StepOne = iota
+	StepTwo
+)
+
+// TaskType of Example.
+const (
+	TaskTypeExample = "Example"
+)
+
+type TaskExample struct{}
```
Comment (zimulala) [inline disttask/example/proto.go:28]: We need to add a comment for the exported type.

File: disttask/example/proto.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2022 PingCAP, Inc.
```
Comment (hawkingrei) [inline disttask/example/proto.go:1]: Please run ```make bazel_prepare``` to update bazel config.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**